pkg/gohai/cpu/cpu_linux_arm64.go

```diff
@@ -116,29 +127,48 @@ func getCPUInfo() (cpuInfo map[string]string, err error) {
 			i++
 		}
 	}
-	cpuInfo["cpu_pkgs"] = strconv.Itoa(len(packages))
-	cpuInfo["cpu_cores"] = strconv.Itoa(len(cores))
-	cpuInfo["cpu_logical_processors"] = strconv.Itoa(len(procCpu))
-	cpuInfo["cache_size_l1"] = strconv.FormatUint(cacheSizes[1], 10)
-	cpuInfo["cache_size_l2"] = strconv.FormatUint(cacheSizes[2], 10)
-	cpuInfo["cache_size_l3"] = strconv.FormatUint(cacheSizes[3], 10)
+	cpuInfo.CPUPkgs = utils.NewValue(uint64(len(packages)))
+	cpuInfo.CPUCores = utils.NewValue(uint64(len(cores)))
+	cpuInfo.CPULogicalProcessors = utils.NewValue(uint64(len(procCPU)))
+	cpuInfo.CacheSizeL1Bytes = utils.NewValue(cacheSizes[1])
+	cpuInfo.CacheSizeL2Bytes = utils.NewValue(cacheSizes[2])
+	cpuInfo.CacheSizeL3Bytes = utils.NewValue(cacheSizes[3])
+
+	// compute total cache size in KB
+	cacheSize := (cacheSizes[1] + cacheSizes[2] + cacheSizes[3]) / 1024
+	cpuInfo.CacheSizeKB = utils.NewValue(cacheSize)
+}
+
+func getCPUInfo() *Info {
+	cpuInfo := &Info{
+		// ARM does not make the clock speed available
+		Mhz: utils.NewErrorValue[float64](utils.ErrNotCollectable),
+	}
 
-	// cache_size uses the format '9216 KB'
-	cpuInfo["cache_size"] = fmt.Sprintf("%d KB", (cacheSizes[1]+cacheSizes[2]+cacheSizes[3])/1024)
+	procCPU, err := readProcCpuInfo()
+	if err == nil {
+		// initialize each field collected from /proc/cpuinfo with a default error
+		cpuInfo.fillProcCPUErr(errors.New("not found in /proc/cpuinfo"))
+		cpuInfo.fillProcCPUInfo(procCPU)
+	} else {
+		cpuInfo.fillProcCPUErr(err)
```

Comment:
Why not setting the error for `cpuInfo.fillProcCPUInfo`?

Reply:
Moved the error handling logic directly inside `fillProcCPUInfo` 👍